### ISSUE TITLE
Sandbox and textures broken

### ISSUE THREAD

**Issue by drsnuggles8** [drsnuggles8/OloEngineBase#20]
Can't build Sandbox anymore, and textures can't be added. Figure out what went bad

**Comment by drsnuggles8** [comment]
Sandbox wasn't building because it was missing shader files, easily fixed.

**Comment by drsnuggles8** [comment]
Now, we investigate when adding textures broke in the Editor. For Sandbox it still works I think.

Go back to PR#10 and see if we can add textures to PinkCube at this point - nope, doesn't work at this point.
Go back to PR#9 and see if we can add textures to PinkCube at this point - yep, works here.

-> PR10 broke it. Check out the history..

What did we change with PR#10?
1. We moved OnDetach, PopLayer and PopOverlay from LayerStack to Application
2. We changed LayerStack to use std::ranges
3. Minor changes to Math.cpp
4. Minor changes to Renderer2D.cpp
5. Minor changes to OpenGLContext.cpp

Which one might fix it? Let's try Renderer2D.cpp stuff first.
Yep, the epsilon comparison was in wrong order. Fixed it.
